src/providers/DynamicProvider.tsx

```diff
@@ -5,12 +5,35 @@ import { EthereumWalletConnectors } from '@dynamic-labs/ethereum';
 import { DynamicContextProvider, useDynamicContext, type UserProfile } from '@dynamic-labs/sdk-react-core';
 import { getAuthToken } from '@dynamic-labs/sdk-react-core';
 import { generateUserSessionDetails } from '../api/graphql-client';
-import { useAuthStore } from '../store/authStore';
+import { type TriggerLoginModal, type TriggerLogout, useAuthStore } from '../store/authStore';
 import { cookies } from '../utils/cookies';
-import { type LoginProviderChildrenProps } from './LoginProvider';
+import type { LoginProviderChildrenProps } from './LoginProvider';
 import { clearStorageByMatchTerm } from '../utils/browser';
 import { decodeAccessToken } from '../utils/token';
 
+type DynamicUtilsProps = {
+  onTriggerLoginModal?: (callback: TriggerLoginModal) => void;
+  onTriggerLogout?: (triggerLogout: TriggerLogout) => void;
+};
+
+const DynamicUtils = ({ onTriggerLoginModal, onTriggerLogout }: DynamicUtilsProps) => {
+  const { sdkHasLoaded, setShowAuthFlow, handleLogOut } = useDynamicContext();
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies(setShowAuthFlow): causes infinite render, probably not memoized
+  useEffect(() => {
```

Comment:
The linter is actually correct here. We missed the callback passed as an argument. So, instead of the ignore we can declare the missing dependency, e.g. setShowAuthFlow.

Do the same for the next ones below please.